extensions/amp-analytics/0.1/variables.js

```diff
@@ -211,6 +211,25 @@ export class VariableService {
     this.register_('LINKER_PARAM', (name, id) =>
       this.linkerReader_.get(name, id)
     );
+
+    // Was set async before
```

Comment:
Thanks for the comment. I think this works for now because `getMacros()` are all used by `expandUrlAsync()` or `expandStringAsync(), please confirm that. 

However I'm a bit concerned about the fact that this won't work well with `expandSync()`. What do you think that we add a `registerSync()` and a `getMacroSync()` method.

Reply:
Discussed offline. 

Recap: This is a valid concern, but since all of the macros are called by expandUrlAsync(), there is not need **yet** to split register and getMacro into a sync and async version. 
I will add comments that will warn of this if we decide to add sync macros in the future.